app/configurator/components/configurator.tsx

```diff
@@ -1,12 +1,279 @@
+import {
+  DataCubeTheme,
+  DataCubeResultOrder,
+  useDataCubesQuery,
+} from "../../graphql/query-hooks";
+import { useLocale } from "../../src";
+import { ChartConfiguratorTable } from "../table/table-chart-configurator";
 import { ChartAnnotationsSelector } from "./chart-annotations-selector";
+import { ChartAnnotator } from "./chart-annotator";
+import { ChartConfigurator } from "./chart-configurator";
 import { ChartOptionsSelector } from "./chart-options-selector";
+import { ChartTypeSelector } from "./chart-type-selector";
 import { DataSetMetadata } from "./dataset-metadata";
-import { PanelLeft } from "./panel-left";
-import { PanelMiddle } from "./panel-middle";
+import { DataSetPreview } from "./dataset-preview";
+import {
+  SearchDatasetBox,
+  SearchFilters,
+  DatasetResults,
+} from "./dataset-search";
+import {
+  PanelLeftWrapper,
+  PanelMiddleWrapper,
+  PanelRightWrapper,
+} from "./layout";
 import { Stepper } from "./stepper";
 
+const useSearchQueryState = () => {
+  const [query, setQuery] = useState<string>("");
+
+  const [order, setOrder] = useState<DataCubeResultOrder>(
+    DataCubeResultOrder.TitleAsc
+  );
+  const previousOrderRef = useRef<DataCubeResultOrder>(
+    DataCubeResultOrder.TitleAsc
+  );
+  const [categoryFilters, setCategoryFilters] = useState<DataCubeTheme[]>([]);
+  const [includeDrafts, setIncludeDrafts] = useState<boolean>(false);
+
+  return {
+    includeDrafts,
+    setIncludeDrafts,
+    onReset: () => {
+      setQuery("");
+      setOrder(previousOrderRef.current);
+    },
+    onTypeQuery: (e: React.ChangeEvent<HTMLInputElement>) => {
+      setQuery(e.currentTarget.value);
+      if (query === "" && e.currentTarget.value !== "") {
+        previousOrderRef.current = order;
+        setOrder(DataCubeResultOrder.Score);
+      }
+      if (query !== "" && e.currentTarget.value === "") {
+        setOrder(previousOrderRef.current);
+      }
+    },
+    query,
+    order,
+    onSetOrder: (order: DataCubeResultOrder) => {
+      previousOrderRef.current = order;
+      setOrder(order);
+    },
+    categoryFilters,
+    onAddFilterTheme: (cat: DataCubeTheme) => {
+      setCategoryFilters(Array.from(new Set([...categoryFilters, cat])));
+    },
+    onRemoveFilterTheme: (cat: DataCubeTheme) => {
+      setCategoryFilters(
+        Array.from(
+          new Set([...categoryFilters.filter((c) => c.iri !== cat.iri)])
+        )
+      );
+    },
+    onToggleFilterTheme: (cat: DataCubeTheme) => {
+      if (categoryFilters.find((c) => c.iri === cat.iri)) {
+        setCategoryFilters(
+          Array.from(
+            new Set([...categoryFilters.filter((c) => c.iri !== cat.iri)])
+          )
+        );
+      } else {
+        setCategoryFilters(Array.from(new Set([...categoryFilters, cat])));
+      }
+    },
+  };
+};
+
+export type SearchQueryState = ReturnType<typeof useSearchQueryState>;
+
+const TextLink = (props: LinkProps) => {
```

Comment:
Nit: Isn't this already available as `<Link variant="inline">`?

Reply:
Good point, done in step 2 👍